.github/workflows/build-images-workflow-run.yml

```
@@ -204,6 +203,7 @@ jobs:
       run-tests: ${{ steps.selective-checks.outputs.run-tests }}
       run-kubernetes-tests: ${{ steps.selective-checks.outputs.run-kubernetes-tests }}
       image-build: ${{ steps.selective-checks.outputs.image-build }}
+      githubRegistry: ${{ steps.determine-github-registry.outputs.githubRegistry }}
```

Having a step and a file for one if is overkill -- there are already too many files in the CI scripts folder, which makes it harder for anyone else to understand it.

This would work wouldn't it?

```suggestion
      githubRegistry: ${{ secrets.GITHUB_REGISTRY || env.GITHUB_REGISTRY }}
```

(and we keep the `GITHUB_REGISTRY` in the env L33-34, and remove the `GITHUB_REGISTRY_SECRET` you added L197)

---

Actually it is even simpler. We can simply do 

```
GITHUB_REGISTRY: ${{ secrets.GITHUB_REGISTRY_OVERRIDE }} 
```

at the top of the build workflow. if the value is empty, the default (old docker package) will be used. No need to use outputs in this workflow at all.